src/DataTypes/DataTypeString.cpp

```diff
@@ -366,12 +377,24 @@ bool DataTypeString::equals(const IDataType & rhs) const
     return typeid(rhs) == typeid(*this);
 }
 
+static DataTypePtr create(const ASTPtr & arguments)
+{
+    if (arguments) {
+        if (arguments->children.size() > 1)
+            throw Exception("String data type family mustnt have more than one argument - size in characters", ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH);
+
+        const auto * argument = arguments->children[0]->as<ASTLiteral>();
+        if (!argument || argument->value.getType() != Field::Types::UInt64 || argument->value.get<UInt64>() == 0)
+            throw Exception("FixedString data type family may have only a number (positive integer) as its argument", ErrorCodes::UNEXPECTED_AST_STRUCTURE);
```

Comment:
This is a String data type family (not FixedString data type family).
And I don't think that's the right solution. I suppose we should treat `VARCHAR(number)` as `FixedString`, not as `String`.

Reply:
For FixedString there are FixedString and BINARY synonim types. 
#10540

Reply:
> I suppose we should treat VARCHAR(number) as FixedString, not as String.

No, that's not right.

Reply:
> > I suppose we should treat VARCHAR(number) as FixedString, not as String.
> 
> No, that's not right.

The `number` in `VARCHAR(number)` means that the size is limited, so `FixedString` seems to be the closest type in ClickHouse to that.

Reply:
The only reason why the size is limited in other DBMS is because no other option is possible.
So, the users tend to write something like `VARCHAR(8000)`. But it still has variable size (unlike `CHAR(10)`).

But `FixedString(8000)` will be extremely inefficient.
Even more, we can treat CHAR(10) as FixedString(10) because the size of FixedString is measured in bytes, not in characters.

Reply:
ok

Reply:
@Potya Could you implement the behaviour that Alexey described here? I mean to treat `CHAR(10)` as `FixedString(10)`?

Reply:
CHAR(x) -> FixedString(x)? but CHAR -> String?

Reply:
Well, it seems a bit inconsistent.
@alexey-milovidov what do you think?

Reply:
No, we cannot say that CHAR is FixedString,
because the size of FixedString is measured in bytes, not characters.

Reply:
@Potya Let's keep your version where `CHAR(x)` is `String` and `CHAR` is `String` too.